### DISCUSSION TITLE
on_event("startup") is called twice

### DISCUSSION THREAD

**Discussion by dbquit27** [fastapi/fastapi#13043] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import APIRouter
from common.utilities import (
    VALID_ENVIRONMENT_VARIABLES
)

router = APIRouter()

@router.on_event("startup")
async def startup_event():
    if VALID_ENVIRONMENT_VARIABLES:
        # function call here. intentionally omitted
        logger.info(
            "App successfully started"
        ) 
    else:
        logger.error(
            "Exception occured: %s", "Required environment variables are not set"
        )
```

### Description

Run the code with the following command:

> uvicorn main:app

The expected result should only log the "App successfully started" once.
However, "App successfully started" is logged twice.

![image](https://github.com/user-attachments/assets/c96be2eb-7f7d-4d74-9a66-84007c68dd64)

This issue does not occur in fastAPI version **0.109.2**.

### Operating System

Windows

### FastAPI Version

0.115.6

### Pydantic Version

2.8.2

### Python Version

3.11.4

**Comment by Kludex** [comment]
You have 2 logging handlers pointing to stdout/stderr.

That's also why you see the other messages duplicated.

**Reply by dbquit27** [reply]
@Kludex , thank you very much for your answer.

Is there a reason why this does not occur in older version of the fastAPI 0.109.2?

**Reply by Kludex** [reply]
Oh, is it happening only on the latest?

**Reply by dbquit27** [reply]
@Kludex 
So far it occurs on versions 0.115.2 and 0.115.6.
But not on 0.109.2.

**Reply by Kludex** [reply]
Does it occur on 0.115.0? When it started?

**Reply by dbquit27** [reply]
I don't know what version it started.
But I can confirm that it also occurs on **0.115.0**.
![image](https://github.com/user-attachments/assets/c20eefd6-70c2-4c09-8b0e-843ce8b278a7)

![image](https://github.com/user-attachments/assets/e98b1bb9-9ec6-4e8d-87a3-548bc894982a)

**Reply by dbquit27** [reply]
Here it is on version **0.109.2**
![image](https://github.com/user-attachments/assets/c164de8d-2cdf-499a-a71d-6a8af3367651)

![image](https://github.com/user-attachments/assets/9fb89066-0892-43f3-9895-7a1b9838cc88)

**Reply by dbquit27** [reply]
Apparently, the issue does not occur until version **0.112.0**
![image](https://github.com/user-attachments/assets/be7563dc-150e-4e30-a053-e5f3b7b91f5a)

![image](https://github.com/user-attachments/assets/ebb2cde0-890a-48c0-9fe1-8d18eef69576)

**Reply by Kludex** [reply]
This version is when fastapi started installing `fastapi-cli` I believe: https://fastapi.tiangolo.com/release-notes/#01120

That may be the problem, and you maybe installing `fastapi[standard]`?

**Reply by dbquit27** [reply]
@Kludex 

> That may be the problem, and you maybe installing fastapi[standard]?

I'm currently not using `fastapi[standard]`

After testing multiple versions, I was able to confirm that the issue started at version **0.112.2**.
